Makefile

```diff
@@ -54,17 +51,28 @@ ifeq ($(GOOS), linux)
 endif
 
 GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
-IMAGE := skopeo-dev$(if $(GIT_BRANCH),:$(GIT_BRANCH))
-# set env like gobuildtag?
-CONTAINER_CMD := ${CONTAINER_RUNTIME} run --rm -i -e TESTFLAGS="$(TESTFLAGS)" #$(CONTAINER_ENVS)
+
+# This is assumed to be set non-empty when operating inside a CI/automation environment
+CI ?=
+
+# This env. var. is interpreted by some tests as a permission to
+# modify local configuration files and services.
+export SKOPEO_CONTAINER_TESTS ?= $(if $(CI),1,0)
+
+# This is a compromise, we either use a container for this or require
+# the local user to have a compatible python3 development environment.
+# Define it as a "resolve on use" variable to avoid calling out when possible
+IMAGE ?= $(shell hack/get_fqin.sh)
+CONTAINER_CMD ?= ${CONTAINER_RUNTIME} run --rm -i -e TESTFLAGS="$(TESTFLAGS)" -e CI=$(CI)
 # if this session isn't interactive, then we don't want to allocate a
 # TTY, which would fail, but if it is interactive, we do want to attach
 # so that the user can send e.g. ^C through.
 INTERACTIVE := $(shell [ -t 0 ] && echo 1 || echo 0)
 ifeq ($(INTERACTIVE), 1)
 	CONTAINER_CMD += -t
 endif
-CONTAINER_RUN := $(CONTAINER_CMD) "$(IMAGE)"
+GOSRC = /src/github.com/containers/skopeo
+CONTAINER_RUN ?= $(CONTAINER_CMD) --security-opt label=disable --privileged -v $(CURDIR):$(GOSRC) -w $(GOSRC) -e SKOPEO_CONTAINER_TESTS=$(SKOPEO_CONTAINER_TESTS) $(IMAGE)
```

Comment:
All but one user sets `SKOPEO_CONTAINER_TESTS=1`; maybe `CONTAINER_RUN` could do that already, and the users could be simplified.

Reply:
okay, fixed.  Thanks.

Reply:
(I still see `SKOPEO_CONTAINER_TESTS=$(SKOPEO_CONTAINER_TESTS)`. If it’s in the next version you will push eventually, that’s perfectly fine, no need to push it right now.)

Reply:
Oh no, I had to un-fix it.  I did it wrong and there was one case where it was passing 0 instead of 1.  TBH, I'm a bit overloaded right now, so let's leave this as a FIXME (I marked it un-resolved).

Reply:
Oh, I got it now.  We can just set `SKOPEO_CONTAINER_TESTS=1` in `$CONTAINER_CMD` and remove it in all the other targets.  When not set (for whatever reason) it will be set to `1` only when `$CI == "true"` (nearly universal standard).

Reply:
Yup!